src/services/dashboards/resource.js

```diff
@@ -947,18 +944,19 @@ export async function resourceData(scopes, skipResources = false, skipTopics = f
     }),
   ]);
 
-  let reports = mergeInResources([], dbData.allReports);
+  let reportsMap = mergeInResources(new Map(), dbData.allReports);
```

Comment:
I'm curious as to whether you tried just using a generic object here, I did some rudimentary googling and it seems like it would be possibly faster (possibly not, was inconclusive)